atc/db/check_lifecycle.go

```diff
@@ -18,13 +18,17 @@ func NewCheckLifecycle(conn Conn) CheckLifecycle {
 
 func (cl *checkLifecycle) DeleteCompletedChecks() error {
 	_, err := cl.conn.Exec(`
-      WITH deleted_builds AS
-      (
+      WITH resource_builds AS (
```

Comment:
I tried to simplify this query a bit and came up with

```
SELECT b.id
FROM builds b
JOIN resources r ON r.id = b.resource_id
WHERE completed
AND b.id < r.build_id;
```

This query is a bit slower on hush-house than the one you provided, though - perhaps with some different indexes things would be a bit faster?

Reply:
Ah yeaah I think if we added an index like `CREATE INDEX deleteme4 ON builds (resource_id, id) WHERE resource_id IS NOT NULL AND completed;`? Not sure how much it would speed up the query. I do know that there is a slow part on the `completed` and `resource_id` is not null filter here https://explain.depesz.com/s/xgPG#l6? Hm wondering if I should test out adding this index to hush-house to see how it affects the performance of this query!

The really unfortunate part is I think it is taking the most time going through the `check_build_events` table and I have no idea on how to speed that part up :(

Reply:
Yeah was just pointing out an alternative! I think the query you have now is good, and as you point out, further optimizations here are probably moot given that they aren’t really the bottleneck